server/jetstream_cluster.go

```diff
@@ -1634,6 +1635,34 @@ func (mset *stream) removeNode() {
 	}
 }
 
+// utility function to return the number of current peers in the specified set,
+// as well as the first current peer and if the leader (always current) is part of the set or not
+func currentPeerCount(ci *ClusterInfo, peerSet []string, leaderId string) (currentCount int, firstPeer string, foundLeader bool) {
+	for _, peer := range peerSet {
+		// selfId is leaderId
+		foundCurrent := peer == leaderId
+		if foundCurrent {
+			foundLeader = true
+		} else {
+			for _, p := range ci.Replicas {
+				if peer == string(getHash(p.Name)) {
```

Comment:
Maybe we could think of storing the hash in PeerInfo - lazily, the first time we detect it is not computed, so we don't have to do that all the time. Don't think it is a problem as-is though, and there are many other places we call getHash()...